### PR TITLE
Stellar: add support for StellarManageBuyOfferOp and StellarPathPaymentStrictSendOp.

### PR DESCRIPTION
**Pull request by overcat**
Change List:
- rename `StellarManageOfferOp` to `StellarManageSellOfferOp`
- rename `StellarPathPaymentOp` to `StellarPathPaymentStrictReceiveOp`
- rename `StellarCreatePassiveOfferOp` to `StellarCreatePassiveSellOfferOp`
- add support for `StellarManageBuyOfferOp`
- add support for `StellarPathPaymentStrictSendOp`
- since the firmware does not support `MuxedAccount`, we refuse to process this type of transaction in trezorctl.
- bump stellar-sdk to 5.0.0, trezorctl is not affected by the breaking changes in it, it's ok to bump it.

Note: This PR is based on the work of @MisterTicot. #627, #579

### CODE REVIEW

**Comment by matejcik** [comment]
just to let you know, this is in the pipeline but I haven't had time to review yet

**Review by matejcik — APPROVED** [review]
very nice, i could only find one typo.

please fix that, rebase on master and fix conflicts

File: core/src/apps/stellar/operations/serialize.py [hunk 1]
```diff
@@ -65,16 +69,28 @@ def write_manage_data_op(w: Writer, msg: StellarManageDataOp) -> None:
...
+    _write_manage_offer_op_common(w, msg)
+
+
+def _write_manage_offer_op_common(
+    w: Writer, msg: StellarManageSellOfferOp | StellarManageBuyOfferOp
+) -> None:
     _write_asset(w, msg.selling_asset)
     _write_asset(w, msg.buying_asset)
-    writers.write_uint64(w, msg.amount)  # amount to sell
+    writers.write_uint64(w, msg.amount)  # amount to sell / sell
```
[the remark this answers is not in the document]
Comment (overcat) [reply core/src/apps/stellar/operations/serialize.py]: Done.

**Review by matejcik — APPROVED** [review]

**Comment by matejcik** [comment]
waiting for the CI pipeline

would you mind doing a corresponding PR to Trezor Connect?

**Comment by overcat** [comment]
> would you mind doing a corresponding PR to Trezor Connect?
@matejcik, I will submit a PR as soon as possible, maybe tomorrow.

**Comment by overcat** [comment]
Hi @matejcik, will the changes here be automatically synced to [trezor-common](https://github.com/trezor/trezor-common)? If not, can you please sync it? Thanks.

**Comment by matejcik** [comment]
@overcat done

**Pull request merged**